scope/src/analyze/cli.rs

```diff
@@ -47,6 +67,34 @@ async fn analyze_logs(found_config: &FoundConfig, args: &AnalyzeLogsArgs) -> Res
     }
 }
 
+async fn analyze_command(found_config: &FoundConfig, args: &AnalyzeCommandArgs) -> Result<i32> {
+    let exec_runner = DefaultExecutionProvider::default();
+
+    let mut command = vec![args.command.clone()];
+    command.extend(args.args.clone());
+    let path = env::var("PATH").unwrap_or_default();
+
+    let capture_opts: CaptureOpts = CaptureOpts {
+        working_dir: &found_config.working_dir,
+        env_vars: Default::default(),
+        path: &path,
+        args: &command,
+        output_dest: OutputDestination::StandardOut,
+    };
+
+    let has_known_error = process_lines(
```

Comment:
Will this stop processing when we see the first error or wait for all output?

Reply:
Good question. This implementation uses [a Cursor](https://github.com/oscope-dev/scope/pull/131/files#r1636576990) to map the fully captured output to a BufReader. This is greedy, and requires the command to finish executing before returning the output.

To get streaming logs from CaptureOutput, I think we'll need a streaming equivalent of `OutputCapture`. Getting the types right on this was a bit beyond me, hence re-using `CaptureOutput`.

After the command has finished, error detection follows the existing pattern of processing each line until all known errors have been accounted for.